apiserver/facades/client/client/backend.go

```diff
@@ -44,6 +44,7 @@ type Backend interface {
 	AllLinkLayerDevices() ([]*state.LinkLayerDevice, error)
 	AllRelations() ([]*state.Relation, error)
 	AllSubnets() ([]*state.Subnet, error)
+	AllSpaces() ([]*state.Space, error)
```

Comment:
The lookup pattern is quite common in the code-base, and we already implement `SpaceLookup` (see the top of this interface).

The advantage in brevity would be to call `AllSpaceInfos` and use `GetByID`. Obviously iterating the spaces is slower than a map lookup, but the number of anticipated spaces in any sane case will make it negligible.

Reply:
I've updated the develop version to us this now.